src/main/java/lotto/domain/result/Rank.java

```diff
@@ -7,31 +7,46 @@
 public enum Rank {
 
     NONE(0, Money.of(0)),
-    FOURTH(3, Money.of(5_000)),
-    THIRD(4, Money.of(50_000)),
-    SECOND(5, Money.of(1_500_000)),
-    FIRST(6, Money.of(2_000_000_000))
+    FIFTH(3, Money.of(5_000)),
+    FOURTH(4, Money.of(50_000)),
+    THIRD(5, Money.of(1_500_000)),
+    SECOND(5, Money.of(30_000_000)),
+    FIRST(6, Money.of(2_000_000_000)),
     ;
 
     private final long matchCount;
     private final Money money;
 
+
     Rank(long matchCount, Money money) {
         this.matchCount = matchCount;
         this.money = money;
     }
 
-    public static Rank of(long matchCount) {
-        return getRank(matchCount);
-    }
+    public static Rank findRankBy(long matchCount, boolean bonusMatch) {
+        if (isSecond(matchCount, bonusMatch)) {
+            return SECOND;
+        }
 
-    public static Rank getRank(long matchCount) {
         return Arrays.stream(values())
-                .filter(r -> r.matchCount == matchCount)
+                .filter(Rank::isNotSecond)
+                .filter(r -> isSameMatchCount(matchCount, r))
                 .findFirst()
                 .orElse(NONE);
     }
 
+    private static boolean isSecond(long matchCount, boolean bonusMatch) {
+        return matchCount == 5 && bonusMatch;
```

Comment:
값을 하드 코딩하는 대신에 이미 정의된 열거형 상수의 값으로 처리해보면 좋을 것 같습니다.
```suggestion
        return matchCount == SECOND.matchCount && bonusMatch;
```

Reply:
네 감사합니다.